mkosi.default.d/10-systemd.conf

```diff
@@ -13,6 +13,7 @@ OutputDirectory=mkosi.output
 [Content]
 BuildDirectory=mkosi.builddir
 Cache=mkosi.cache
+SourceFileTransfer=mount
```

Comment:
hmm?

Reply:
The kernel source tree is huge, it's quite slow to copyitin for builds all the time. It's also in .gitignore so it wouldn't be available unless we use SourceFileTransfer mount

Reply:
hmm, what's the rationale, why didn't we always do this via mounts?

Reply:
hmm, does mkosi even user nspawns's userns mode?

but yeah, I guess mkosi must assume people have the same build and source tree, and then the perms thing does become a mess indeed. Given we have that neatly separate I figure we should just use mounting.

Can you make that part a separate commit, though? I think this makes sense anyway, regardless of the kernel stuff.

Reply:
@keszybz I think you added the original code to mkosi for copying sources files. Do you see any problem with switching to mount mode for source transfer for systemd itself?

Reply:
(maybe we should add docs to mkosi's SourceFileTransfer= setting making clear that "mount" is actually recommended if you have a neatly separate build and source tree)